docs/source/examples/pt_zne.md

```diff
@@ -0,0 +1,406 @@
+            return CircuitOperation(
+                Circuit(
+                    op.gate.on(*op.qubits), 
+                    Ry(rads=rads).on_each(op.qubits),
+                ).freeze())
+        return op
+
+    return GateSubstitutionNoiseModel(noisy_c_gate)
+
+def execute(circuit: Circuit, noise_level: float):
+    """Returns Tr[ρ |0⟩⟨0|] where ρ is the state prepared by the circuit."""
+    return (
+        DensityMatrixSimulator(noise=get_noise_model(noise_level=noise_level))
+        .simulate(circuit)
+        .final_density_matrix[0, 0]
+        .real
+    )
+
+
+# Set the intensity of the noise
+NOISE_LEVEL = 0.2
+
+
+# Compute the expectation value of the |0><0| observable
+# in both the noiseless and the noisy setup
+ideal_value = execute(circuit, noise_level=0.0)
+noisy_value = execute(circuit, noise_level=NOISE_LEVEL)
+
+NUM_TWIRLED_VARIANTS = 300
+twirled_circuits = generate_pauli_twirl_variants(circuit, num_circuits=NUM_TWIRLED_VARIANTS)
+
+# Average results executed over twirled circuits
+from functools import partial
+from mitiq import Executor
+pt_vals = Executor(partial(execute, noise_level=NOISE_LEVEL)).evaluate(twirled_circuits)
+twirled_result = np.average(pt_vals)
+
+
+print(f"Error without twirling: {abs(ideal_value - noisy_value) :.3}")
+print(f"Error with twirling: {abs(ideal_value - twirled_result) :.3}")
+```
+
+```{caution}
+It is worth noting that Pauli twirling's goal is to tailor the noise from coherent to incoherent. 
+
+Depending on the noise strength, type of coherent noise etc. this transformation might not give better results after the Pauli twirled circuit is executed. See the plot in the next section for an example.
+```
+
+## Combining Pauli Twirling with ZNE
+
+```{code-cell} ipython3
+from mitiq.zne import execute_with_zne
+
+executor=partial(execute, noise_level=NOISE_LEVEL)
+zne_pt_vals = []
+
+for twirled_circuit in twirled_circuits:
+    zne_pt_vals.append(execute_with_zne(twirled_circuit, executor))
+
+mitigated_result = np.average(zne_pt_vals)
+
+print(f"Error without twirling: {abs(ideal_value - noisy_value) :.3}")
+print(f"Error with twirling: {abs(ideal_value - twirled_result) :.3}")
+print(f"Error with ZNE + PT: {abs(ideal_value - mitigated_result) :.3}")
+
+```
+Accordingly, depending on the noise strength, a combination of PT and ZNE do not work that well compared to just PT or ZNE. Thus, it is important to understand when combining a noisy tailoring technique with an error mitigation technique provides a significant advantage. 
+
+```{code-cell} ipython3
+
+import matplotlib.pyplot as plt
+
+# Plot error vs noise strength
+noise_strength = np.linspace(0.0, 1.0, 50)
+error_no_zne_no_twirling = []
+error_with_twirling = []
+error_with_twirling_and_zne = []
+zne_vals = []
+ideal_values = []
+NUM_TWIRLED_VARIANTS = 30
+
+for strength in noise_strength:
+    ideal_value = execute(circuit, noise_level=0.0)
+    ideal_values.append(ideal_value)
```

Comment:
These go outside the for loop

Reply:
Not really. These lines are making sure the size of `ideal_values` is the same as the size of the other values we are comparing it to.

Reply:
Never mind. I changed this.